File: Python/basics/custom_exception.py
class age_exception(Exception):
    def __init__(self,msg="enter a valid age"):
        super().__init__(msg)
try:
    age = int(input("enter age: "))
    if(age<18 or age>100):
        raise age_exception
    print(age)
except age_exception as e:
    print(e)

File: training/java/core/pack/Day5/CustomException/CustomException.java
package Day5.CustomException;
import java.util.*;

public class CustomException {
	
public static void main(String args[]) throws InvalidInputException  {
	Scanner sc=new Scanner(System.in);
	System.out.println("Enter the string that contain only alphabets");
	String s=sc.next();
	
	try {
	if(!s.matches("[a-zA-Z]")) 
		throw new InvalidInputException("invalid input");
	
	
	System.out.println("You have entered "+s);
	
}
	catch(InvalidInputException e) {
		System.out.println("Only alphabets are allowed");
	}
}
}


class InvalidInputException extends Exception{
	public InvalidInputException(String message){
		super(message);
	}
}
